npy_array/data_type.h

```diff
@@ -0,0 +1,81 @@
+#ifndef NPY_ARRAY_DATA_TYPE_H_
+#define NPY_ARRAY_DATA_TYPE_H_
+
+#include <cstdint>
+
+#include "npy_array/half.h"
+
+namespace npy_array {
+
+// Underlying data type used by pixel formats.
+enum class DataType {
+  kUndefined,
+
+  kInt8,
+  kInt16,
+  kInt32,
+  kInt64,
+
+  kUint8,
+  kUint16,
+  kUint32,
+  kUint64,
+
+  kFloat16,
+  kFloat32,
+  kFloat64,
+};
+
+template <typename Sink>
+void AbslStringify(Sink& sink, DataType dt) {
+  switch (dt) {
+    case DataType::kUndefined:
+      sink.Append("kUndefined");
+      break;
+    case DataType::kInt8:
+      sink.Append("kInt8");
+      break;
+    case DataType::kInt16:
+      sink.Append("kInt16");
+      break;
+    case DataType::kInt32:
+      sink.Append("kInt32");
+      break;
+    case DataType::kInt64:
+      sink.Append("kInt64");
+      break;
+
+    case DataType::kUint8:
+      sink.Append("kUint8");
+      break;
+    case DataType::kUint16:
+      sink.Append("kUint16");
+      break;
+    case DataType::kUint32:
+      sink.Append("kUint32");
+      break;
+    case DataType::kUint64:
+      sink.Append("kUint64");
+      break;
+    case DataType::kFloat16:
+      sink.Append("kFloat16");
+      break;
+    case DataType::kFloat32:
+      sink.Append("kFloat32");
+      break;
+    case DataType::kFloat64:
+      sink.Append("kFloat64");
+      break;
+  }
+}
+
+// Returns the size of 1 element of type `dt`, in bytes.
+size_t elementSize(DataType dt);
```

Comment:
@yyuting If you prefer, I can make all function names `Capital`case to match Google's code style.

Or `snake_case` to match that of nda. Since we started with `npy_array`, probably the former?

Reply:
I slightly prefer to keep it as is for minimum code change between here and our own repo. But capital case also sounds fine.

Reply:
Probably `CapitalCase` for Google-style consistency. At least it's two, not three.

Reply:
Ok, this is done. I made most function as `CapitalCase` to match Google (which the existing serialize/deserialize functions use).

I left the ones that have the same name as `nda::array` as is, along with functions like `size()` and `empty()` that match STL.

Also matching Google style are accessors that return the value. As in `dims() { return dims_; }`